plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -70,6 +93,8 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
                     .build();
         }
 
+        Map<Integer, Long> ndvs = readNdvs(session, icebergTable);
```

Comment:
Is there any particular reason why  ndvs are read so far away from the place where it is read ?